File: Randompack.jl/examples/ThreadSpawn.jl
using Randompack
using Base.Threads

function worker(i, results)
  rng = rng_create()
  rng_seed!(rng, 123; spawn_key=[i-1])
  results[i] = random_normal(rng, 1000)
end

results = Vector{Vector{Float64}}(undef, 3)
tasks = Vector{Task}(undef, 3)
for i in 1:3
  tasks[i] = @spawn worker(i, results)
end
foreach(fetch, tasks) # wait till tasks finish
println(results)

File: python/examples/ThreadSpawn.py

import numpy as np, randompack
from threading import Thread

def worker(i, results):
  rng = randompack.Rng()
  rng.seed(123, spawn_key=[i])
  results[i] = rng.normal(1000)

results = [None]*3
threads = [None]*3
for i in range(3):
  threads[i] = Thread(target=worker, args=(i,results))
  threads[i].start()
for thread in threads: thread.join() # wait till threads finish
print(np.array(results))
